### ISSUE TITLE
Docs outdated about DRF not passing request to authenticate

### ISSUE THREAD

**Issue by mscansian** [jazzband/django-axes#353]
I've noticed that a claim in the documentation might be outdated. When discussing the issues with authentication, it blames the DRF as it does not pass the request to the `authenticate` function. This might be true in the past, but I think [it's not the case anymore](https://github.com/encode/django-rest-framework/blob/master/rest_framework/authentication.py#L97) ([commit](https://github.com/encode/django-rest-framework/commit/161dc2df2ccecc5307cdbc05ef6159afd614189e)), as it was fixed in version `3.7.0`.

https://django-axes.readthedocs.io/en/latest/configuration.html#authentication-backend-problems
`Notably Django Rest Framework (DRF) BasicAuthentication does not pass request. Here is an example workaround for DRF.`

Can anybody confirm this, so I can remove this comment from the docs? Thanks.

**Comment by aleksihakli** [comment]
It seems like the request is passed. Since we have [fairly strong check for the correct request passing behaviour](https://github.com/jazzband/django-axes/blob/master/axes/backends.py#L33) I think the docs can be updated :)

**Comment by wodCZ** [comment]
The example workaround is still relevant for other libraries. For example [drf-jwt serializer](https://github.com/GetBlimp/django-rest-framework-jwt/blob/master/rest_framework_jwt/serializers.py#L44) has to be pathed (with which the example did help me).
Also, some users may still use older versions of DRF.

Maybe rewording the sentence to "blame older versions of DRF" would be most useful for users of the documentation?

**Comment by camilonova** [comment]
@wodCZ we are always open to pull requests :)

**Comment by markddavidoff** [comment]
I'm happy to hear this is now fixed in DRF! I've opened a PR @ https://github.com/jazzband/django-axes/pull/357.

**Comment by markddavidoff** [comment]
Closing as this is addressed in the PR
